context.go

```diff
@@ -265,6 +265,18 @@ func (c *Context) Get(key string) (value any, exists bool) {
 	return
 }
 
+// GetAs returns the value for the given key, ie: (value, true).
+// If the value does not exist it returns (nil, false)
+func (c *Context) GetAs[T any](key string) (result T, exists bool) {
```

Comment:
We also need support go1.16 up version.

Reply:
This will be problematic, as Generics were introduced in Go  1.19. Any suggestions?

Reply:
What's the benefit of this method over the non-generic version? The biggest issue with `Get` is that a failed type assertion causes a run-time panic, which seems to be the same here?

Reply:
No need to assert the variable to a certain type. The assertion is also checked and returns true/false depending on the case.

Reply:
@appleboy gin requires go 1.21 or above now. Would you consider this feature again??

Reply:
@laggu Yes, we should reconsider the changes.

Reply:
@appleboy any progress???

Reply:
https://github.com/gin-gonic/gin/pull/4080

Reply:
The changing is not with exported generic function which is not for general use
Why not you choosing this pr?? 
Exported generic function is much useful for develoepr who use gin

Reply:
@laggu just refer to the PR #4080